libs/ml/include/ml/ops/strided_slice.hpp

```diff
@@ -0,0 +1,352 @@
+  using ArrayPtrType  = std::shared_ptr<TensorType>;
+  using VecTensorType = typename Ops<T>::VecTensorType;
+  using SPType        = OpStridedSliceSaveableParams<T>;
+  using MyType        = StridedSlice<TensorType>;
+
+  explicit StridedSlice(SizeVector const &begins, SizeVector const &ends,
+                        SizeVector const &strides = {})
+    : begins_(begins)
+    , ends_(ends)
+    , strides_(strides)
+
+  {
+    assert(begins.size() == ends.size());
+    if (strides.empty())
+    {
+      strides_ = begins;
+      for (SizeType i{0}; i < strides.size(); i++)
+      {
+        strides_.at(i) = 1;
+      }
+    }
+  }
+
+  explicit StridedSlice(SPType const &sp)
+    : Ops<T>(sp)
+  {
+    begins_  = sp.begins;
+    ends_    = sp.ends;
+    strides_ = sp.strides;
+  }
+
+  ~StridedSlice() override = default;
+
+  std::shared_ptr<OpsSaveableParams> GetOpSaveableParams() override
+  {
+    auto sp = std::make_shared<SPType>();
+
+    sp->begins  = begins_;
+    sp->ends    = ends_;
+    sp->strides = strides_;
+
+    return sp;
+  }
+
+  std::shared_ptr<fetch::ml::ops::Ops<TensorType>> MakeSharedCopy(
+      std::shared_ptr<fetch::ml::ops::Ops<TensorType>> me) override
+  {
+    FETCH_UNUSED(me);
+    assert(me.get() == this);
+
+    return std::make_shared<MyType>(*this);  // calls default copy constructor of MyType;
+  }
+
+  void Forward(VecTensorType const &inputs, TensorType &output) override;
+
+  std::vector<TensorType> Backward(VecTensorType const &inputs,
+                                   TensorType const &   error_signal) override;
+
+  SizeVector ComputeOutputShape(VecTensorType const &inputs) const override
+  {
+
+    SizeVector output_shape = inputs.front()->shape();
+
+    // Calculate number of stride size steps from specified begin to specified end for each
+    // dimension
+    for (SizeType i{0}; i < begins_.size(); i++)
+    {
+      assert(strides_.at(i) != 0);
+      assert(begins_.at(i) <= ends_.at(i));
+      output_shape.at(i) = ((ends_.at(i) - begins_.at(i)) / strides_.at(i)) + 1;
+    }
+
+    return output_shape;
+  }
+
+  SizeVector begins_;
+  SizeVector ends_;
+  SizeVector strides_;
+
+  static constexpr OpType OpCode()
+  {
+    return OpType::OP_STRIDED_SLICE;
+  }
+
+  static constexpr char const *DESCRIPTOR = "StridedSlice";
+};
+
+/**
+ * Forward pass is done by assigning values in given ranges with stride size step for every
+ * dimmension from larger input tensor to smaller output tensor.
+ * @tparam T
+ * @param inputs
+ * @param output
+ */
+template <class T>
+void StridedSlice<T>::Forward(VecTensorType const &inputs, TensorType &output)
+{
+  assert(inputs.size() == 1);
+  assert(output.shape() == this->ComputeOutputShape(inputs));
+
+  switch (inputs.at(0)->shape().size())
+  {
+    // 1D Tensor
+  case 1:
+  {
+    for (SizeType i{0}; i < output.shape().at(0); i++)
+    {
+      output.At(i) = inputs.at(0)->At(begins_.at(0) + i * strides_.at(0));
```

Comment:
I don't understand why you use .At(i), .At(i, j) etc. here.
Pretty sure you can do all this a lot better with iterators.
It looks like TensorSliceIterator has a constructor that takes stride already, so if you add another Slice method to tensor I think the strided slice implementation can be handled there.